client/src/pages/User/EditProfile/EditProfile.tsx

```diff
@@ -4,87 +4,317 @@ import Button from '@material-ui/core/Button';
 import Box from '@material-ui/core/Box';
 import Paper from '@material-ui/core/Paper';
 import TextField from '@material-ui/core/TextField';
-<<<<<<< HEAD
 import Typography from '@material-ui/core/Typography';
 import SideNav from '../../../components/SideNav/SideNav';
 import AuthMenu from '../../../components/AuthMenu/AuthMenu';
-=======
-import { Typography } from '@material-ui/core';
-import { Link } from 'react-router-dom';
-import { useState } from 'react';
->>>>>>> 52eb0b573acece8fbff2ea22f1a0d6ded488c989
+import { useEffect, useState } from 'react';
+import { User } from '../../../interface/User';
+import { FetchOptions } from '../../../interface/FetchOptions';
+import { Formik, FormikHelpers, Field } from 'formik';
+import * as Yup from 'yup';
+import { CircularProgress } from '@material-ui/core';
 
-export default function EditProfile(): JSX.Element {
+interface Props {
+  loggedInUser: User;
+}
+
+export default function EditProfile({ loggedInUser }: Props): JSX.Element {
+  const profile = loggedInUser.profile;
   const classes = useStyles();
+  const [loading, setLoading] = useState(true);
+  const [profileData, setProfileData] = useState({
+    id: ' ',
+    firstName: ' ',
+    lastName: ' ',
+    gender: ' ',
+    email: ' ',
+    phoneNumber: ' ',
+    address: ' ',
+    description: ' ',
+  });
+  const getProfile = async (profile: string) => {
+    const fetchOptions: FetchOptions = {
+      method: 'GET',
+      headers: { 'Content-Type': 'application/json' },
+      credentials: 'include',
+    };
+    await fetch(`/profile/${profile}`, fetchOptions)
+      .then((res) => res.json())
+      .then((data) => {
+        const temp = data.success.profile;
+        setProfileData({
+          id: temp.id,
+          firstName: temp.firstName,
+          lastName: temp.lastName,
+          gender: temp.gender,
+          email: temp.email,
+          phoneNumber: temp.phoneNumber,
+          address: temp.address,
+          description: temp.description,
+        });
+        setLoading(false);
+      })
+      .catch(() => ({
+        error: { message: 'Unable to connect to server. Please try again' },
+      }));
+  };
```

Comment:
This function seems to go out of the scope of its definition. It is rather also calling `setProfileData` instead of just `getProfile`. Well, it's just a good practice to name a function to reflect exactly what it does.